vnext/ReactUWP/Utils/ResourceBrushUtils.cpp

```diff
@@ -0,0 +1,141 @@
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOff, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOffPointerOver, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOffPressed, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOffDisabled, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOn, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOnPointerOver, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOnPressed, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOnDisabled, brush);
+}
+
+void UpdateToggleSwitchThumbResourceBrushes(
+    const winrt::ToggleSwitch &toggleSwitch,
+    const winrt::Brush thumbBrush) {
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOff, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOffPointerOver, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOffPressed, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOffDisabled, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOn, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOnPointerOver, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOnPressed, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOnDisabled, thumbBrush);
+}
+
+void UpdateToggleSwitchTrackResourceBrushes(
+  const winrt::ToggleSwitch& toggleSwitch,
+  const winrt::Brush onTrackBrush,
+  const winrt::Brush offTrackBrush) {
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOn, onTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOnPointerOver, onTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOnPressed, onTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOnDisabled, onTrackBrush);
+
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOff, offTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOffPointerOver, offTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOffPressed, offTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOffDisabled, offTrackBrush);
+}
+
+bool IsObjectATextControl(const winrt::DependencyObject &object) {
+  return object.try_as<winrt::TextBox>() != nullptr || object.try_as<winrt::PasswordBox>() != nullptr ||
+      object.try_as<winrt::RichEditBox>() != nullptr || object.try_as<winrt::AutoSuggestBox>() != nullptr;
+}
+
+void UpdateControlBackgroundResourceBrushes(
+    const winrt::Windows::UI::Xaml::FrameworkElement &element,
+    const winrt::Media::Brush brush) {
+  if (IsObjectATextControl(element)) {
+    UpdateTextControlBackgroundResourceBrushes(element, brush);
+  }
+}
+
+void UpdateControlForegroundResourceBrushes(
+    const winrt::Windows::UI::Xaml::DependencyObject object,
+    const winrt::Media::Brush brush) {
+  if (IsObjectATextControl(object)) {
```

Comment:
I'm happy to rename it but I'm not sure "TextEdit" adds clarity. In the example you pointed out, VirtualTextViewManager calls TryUpdateForeground with a `TextElement` (not TextEdit), and as a result the Foreground DP gets set, but none of the Resources do, because IsObjectATextControl will return false; `Windows.UI.Xaml.Documents.TextElement` don't use these Resource keys (in fact, as types that don't derive from FrameworkElement, they don't even have their own `Resources` DP). I'm hoping that the naming, `IsObjectATextControl` sets the expectation that it would be true for only TextBox, PasswordBox, RichEditBox, and AutoSuggestBox.

To explain my naming choice, in case that helps, I was using "TextControl" to be consistent with (1) the docs.microsoft.com documentation*, and (2) with the name of the shared Resource brush values, e.g. "**TextControl**Foreground".

*"This table shows the resources used by the TextBox control. Resources that start with "TextControl" are shared by TextBox, PasswordBox, RichEditBox, and AutoSuggestBox."

Reply:
@ddalp let me know what you think; I'll rename as desired. Hoping to merge today if that's possible

Reply:
That's fine.. I just want to make sure we remove the TryUpdateResourceBrush, it is there because Text element passed in as DependencyObject instead of FrameworkElement.

---
In reply to: [346047250](https://github.com/microsoft/react-native-windows/pull/3652#discussion_r346047250) [](ancestors = 346047250)